### PR TITLE
Readable Code for Audio Signal Visualization

### PR DESCRIPTION
**Pull request by lucifertrj**
The following code takes to many variables, and I have simplified the following code into more readable format. 
```py
rows = 3
cols = 3
n = rows * cols
fig, axes = plt.subplots(rows, cols, figsize=(16, 9))

for i in range(n):
  if i>=n:
    break
  r = i // cols
  c = i % cols
  ax = axes[r][c]
  ax.plot(example_audio[i].numpy())
  ax.set_yticks(np.arange(-1.2, 1.2, 0.2))
  label = label_names[example_labels[i]]
  ax.set_title(label)
  ax.set_ylim([-1.1,1.1])
```
to
```py
plt.figure(figsize=(16,10))
for i in range(9):
  plt.subplot(3,3,i+1)
  audio_signal = example_audio.numpy()[i]
  plt.plot(audio_signal)
  plt.title(label_names[example_labels[i]])
  plt.yticks(np.arange(-1.2, 1.2, 0.2))
  plt.ylim([-1.1,1.1])
```

### CODE REVIEW

**Comment by markmcd** [comment]
> The following code takes to many variables and I have simplified the following code into more readable format.

Thanks - I like your suggestion, but some of these variables are there to improve readability, so I have some requests:

* Leave the `rows`, `cols` and `n` assignments, and make sure they work. It's consistent with the rest of the document, is clearer, and allows readers to tweak to fit their screen.
* Leave `example_audio[i].numpy()` (or just `example_audio[i]` should work in this context)
* Add spaces after every comma

**Comment by lucifertrj** [comment]
Hi @markmcd, 

Thank you for the review. By code readable, I mainly meant simplifying  `r = i // cols` and  `c = i % cols` with `plt.subplot()`. I have kept `n`, `rows` and `cols`. And I have made sure that it works. Here is the result: 

Screenshot from 2023-02-21 23-36-38

**Comment by lucifertrj** [comment]
> Hi @markmcd,
> 
> Thank you for the review. By code readable, I mainly meant simplifying `r = i // cols` and `c = i % cols` with `plt.subplot()`. I have kept `n`, `rows` and `cols`. And I have made sure that it works. Here is the result:
> 
> Screenshot from 2023-02-21 23-36-38

Also no need to call `numpy()`, which is mentioned in the original as well as updated code.

**Comment by lucifertrj** [comment]
Also, no need to mention `numpy()`. This is included in both the code i.e., official docs and my previous commit. I have fixed it in latest commit.

File: site/en/tutorials/audio/simple_audio.ipynb [hunk 1]
```diff
@@ -1,987 +1,1026 @@
...
+    "id": "8yuX6Nqzf6wT"
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(16, 10))\n",
+    "rows = 3\n",
+    "cols = 3\n",
+    "n = rows * cols\n",
+    "for i in range(n):\n",
+    "  plt.subplot(3, 3, i+1)\n",
```
Comment (MarkDaoust) [inline site/en/tutorials/audio/simple_audio.ipynb:363]:
```suggestion
    "  plt.subplot(rows, cols, i+1)\n",
```

Comment (lucifertrj) [reply site/en/tutorials/audio/simple_audio.ipynb:363]: sorry to miss this out. I shall update it

**Review by MarkDaoust — DISMISSED** [review]

**Comment by lucifertrj** [comment]
@MarkDaoust, I updated (3, 3) to (rows, cols). Sorry, I didn't notice that you already reviewed the PR

**Comment by 8bitmp3** [comment]
@lucifertrj I formatted the notebook for you again, given the large diff because of the meta data.

TODO for you:

1) Please remove the cell outputs, and then
2) Format the notebook again.

This guide here explains how apply the `tensorflow-docs` `nbfmt` tool to any IPYNB file to make sure the JSON is cleaner, and passes the tests: https://www.tensorflow.org/community/contribute/docs#notebook_formatting

TL;DR suggested steps:

1. In a separate Python environment (with `venv`/`virtualenv`), or in the same one where you have TensorFlow installed, install the `tensorflow-docs` package: `python3 -m pip install git+https://github.com/tensorflow/docs`.
2. Run: `python3 -m tensorflow_docs.tools.nbfmt simple_audio.ipynb`

The output should then say:

```
Format notebook: simple_audio.ipynb
```

Then you can `git` add, commit, and push to your branch.

If you have questions, feel free to ask :+1:

**Comment by lucifertrj** [comment]
> @lucifertrj I formatted the notebook for you again, given the large diff because of the meta data.
> 
> TODO for you:
> 
> 1. Please remove the cell outputs, and then
> 2. Format the notebook again.
> 
> This guide here explains how apply the `tensorflow-docs` `nbfmt` tool to any IPYNB file to make sure the JSON is cleaner, and passes the tests: https://www.tensorflow.org/community/contribute/docs#notebook_formatting
> 
> TL;DR suggested steps:
> 
> 1. In a separate Python environment (with `venv`/`virtualenv`), or in the same one where you have TensorFlow installed, install the `tensorflow-docs` package: `python3 -m pip install git+https://github.com/tensorflow/docs`.
> 2. Run: `python3 -m tensorflow_docs.tools.nbfmt simple_audio.ipynb`
> 
> The output should then say:
> 
> ```
> Format notebook: simple_audio.ipynb
> ```
> 
> Then you can `git` add, commit, and push to your branch.
> 
> If you have questions, feel free to ask +1

Hi @8bitmp3, I apologise for taking a two days gap. I was busy with my internship work. I will complete the TODO assigned to this PR tomorrow without fail.

**Comment by lucifertrj** [comment]
Done @8bitmp3 
Screenshot from 2023-02-25 19-51-53

**Comment by lucifertrj** [comment]
@markmcd 

The following code does nothing in the simple audio documentation, and train_ds.take(1) doesn't require a break. Because this will return just a single batch/ 

```py
for example_spectrograms, example_spect_labels in train_spectrogram_ds.take(1):
  break
```
The above code can be modified into the following:
```py
for example_spectrograms, example_spect_labels in train_spectrogram_ds.take(1):
  print(example_spectrograms.shape)
  print(example_spect_labels.shape)
```
Should I make the changes, let me know what you think.

**Comment by markmcd** [comment]
> The following code does nothing in the simple audio documentation, and train_ds.take(1) doesn't require a break. Because this will return just a single batch/
> 
> ```python
> for example_spectrograms, example_spect_labels in train_spectrogram_ds.take(1):
>   break
> ```

This code does do something - try removing it, restarting and running the notebook to see what happens. `.take(1)` returns a dataset, and the for-loop calls its `__iter__` method, which materialises rows from the dataset.

You can see in the next cell that `example_spectrograms` is used. Without this loop, that variable would not exist.

Another way this could be written that might avoid this confusion is:

```python
example_spectrograms, example_spect_labels = next(train_spectrogram_ds.take(1))
```

This is pretty explicit about what's happening, but `next()` may be a little harder to grok than a for-loop for someone newer to Python.

**Comment by lucifertrj** [comment]
> to grok than a for-loop for someone newer to Python.

Got it thanks

**Comment by lucifertrj** [comment]
@8bitmp3 , the PR is still showing `Code Owner Review required`. I have made the necessary changes.

**Review by MarkDaoust — APPROVED** [review]

**Review by 8bitmp3 — APPROVED** [review]
